### PR TITLE
DOC: Clarify wording on Copies and Views page

### PR DESCRIPTION
**Pull request by davidgilbertson**
The page: https://numpy.org/doc/stable/user/basics.copies.html#other-operations

I read this and assumed that if `b` is a view of `a`, then `b.base is a` would be `True`.
```py
>>> a = np.arange(4).reshape(2, 2)
>>> b = a[:]
>>> b.base
array([0, 1, 2, 3])
```

So the wording "returns **the originary array**" is a bit misleading IMO.

### CODE REVIEW

**Comment by seberg** [comment]
Hmmm, I wonder if this is good advice even broader speaking.  The problem is not that it is incorrect (if `base` is not `None` we do have a view).  But just because a function returned a view, does not mean it is not a view into a copy.

```
arr = np.ones(10)
np.nonzero(arr)[0].base is None

# or:
arr = np.ones((4, 5))
arr[:, [0, 1]].base is None
``` 
The first is maybe not the best example since it clearly returns a new array of indices.  But the second clearly is a copy (advanced indexing just like the example!), but more is going on behind the scenes so that the result is a view into a fresh copy.
From a user perspective, there is no reason to worry about that though, they can act as if they got a copy.

Long story short, I don't have a good idea right now, but maybe it would be good to also mention `np.may_share_memory()` somewhere in the paragraph?  I don't think the general advice is bad, but it does have some limitations.

Interestingly, this was probably written at a time where the first was correct for this example (NumPy drops unnecessary `.base` chains, but that was not always the case.)

That said, the change is probably a small improvement, so I would be fine to just adding it.

**Review by rossbar — COMMENTED** [review]
IMV the original wording is correct. I think the confusion in this case arises from the fact that the array returned by `.reshape` is itself a view (with modified shape/strides) of the original array returned by arange. If you decompose the steps a bit this becomes more clear:

```python
>>> a = np.arange(4)
>>> a_rs = a.reshape(2, 2)
>>> a.base

>>> a_rs.base is a
True
>>> b = a_rs[:]
>>> b.base is a_rs  # No, because a_rs is itself a view
False
>>> b.base is a  # The "original" array
True
```

More generally I agree with @seberg - relying on `.base is ` is fragile as there may be some under-the-hood operations that create arrays or views themselves.

Overall I'm -1 on this specific wording change, but agree that the best-practices surrounding `base` are not particularly well-documented.

**Comment by davidgilbertson** [comment]
Oh interesting, so with `b` as a view of `a`, it is true that `b.base` "returns the originary array" but it is **not necessarily true** that `a` itself "returns the originary array" if you use the same definition of "original array".

I suspect that most readers (presumably reading the page to learn how copies and view work) would assume that "original array" refers to `a`.

So I guess this can be summarised as "technically correct but probably misleading".

If that's acceptable feel free to close this.

**Comment by rossbar** [comment]
I will close this, especially since reading a bit further, the next example immediately following this wording explains that `reshape` creates a view. Thanks for the suggestions @davidgilbertson
